presto-main/src/main/java/io/prestosql/Session.java

```diff
@@ -351,7 +351,11 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
                 queryId,
                 Optional.of(transactionId),
                 clientTransactionSupport,
-                new Identity(identity.getUser(), identity.getPrincipal(), roles.build(), identity.getExtraCredentials()),
+                Identity.forUser(identity.getUser())
```

Comment:
Add a method for making a builder from existing identity:

`Identity.fromIdentity(identity)`